### PR TITLE
Implement SDL 0179 - Pixel Density and Scale

### LINKED ISSUE #1007: [SDL 0179] Pixel density and Scale
Proposal: [Pixel density and Scale](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0179-pixel-density-and-scale.md)

> This proposal is about adding parameters to `VideoStreamingCapabilities` for pixel density and scale.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/526

### PR DESCRIPTION
**Pull request by NicoleYarroch**
Fixes #1007

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit tests added for:
- [x] SDLStreamingVideoLifecycleManagerSpec
- [x] SDLVideoStreamingCapabilitySpec
- [x] SDLTouchManagerSpec
- [x] SDLHapticManagerSpec

### Summary
* New parameters, `diagonalScreenSize`, `pixelPerInch`, and `scale`, were added to `SDLVideoStreamingCapability`.
* The `SDLH264VideoEncoder`'s `dimensions` are now scaled in the `SDLStreamingVideoLifecycleManager`.
* The view controller frame in `SDLCarWindow` is now set to the scaled video resolution sent by core in the `videoStreamingCapability`. 
* The `SDLTouchManager` now scales the touch event coordinates returned by core.
* The `SDLFocusableItemLocator` now scales the haptic data sent to core.

### Contributors
Based on PR #1401
Contributors: [@lnafarrateluxoft](https://github.com/lnafarrateluxoft),  [@yLeonid](https://github.com/yLeonid)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m [hunk 1]
```diff
@@ -583,37 +595,39 @@ __block void (^sendNotificationForHMILevel)(SDLHMILevel hmiLevel, SDLVideoStream
...
                     beforeEach(^{
+                        streamingLifecycleManager.preferredResolutions = @[];
+                        
                         testVideoStartServicePayload = [[SDLControlFramePayloadVideoStartServiceAck alloc] initWithMTU:testMTU height:SDLControlFrameInt32NotFound width:SDLControlFrameInt32NotFound protocol:nil codec:nil];
                         testVideoMessage = [[SDLV2ProtocolMessage alloc] initWithHeader:testVideoHeader andPayload:testVideoStartServicePayload.data];
+                        expect(@(CGSizeEqualToSize(streamingLifecycleManager.screenSize, CGSizeZero))).to(equal(@YES));
                     });
-
-                    context(@"If the data source is nil", ^{
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m]:
```suggestion
```

File: SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m [hunk 2]
```diff
@@ -583,37 +595,39 @@ __block void (^sendNotificationForHMILevel)(SDLHMILevel hmiLevel, SDLVideoStream
...
-
-                    context(@"If the data source is nil", ^{
+                    
+                    context(@"If no preferred resolutions were set in the data source", ^{
                         beforeEach(^{
                             streamingLifecycleManager.dataSource = nil;
                             [streamingLifecycleManager handleProtocolStartServiceACKMessage:testVideoMessage];
                         });
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m]:
```suggestion

```

File: SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m [hunk 3]
```diff
@@ -583,37 +595,39 @@ __block void (^sendNotificationForHMILevel)(SDLHMILevel hmiLevel, SDLVideoStream
...
                         });
-
+                        
                         it(@"should not replace the existing screen resolution", ^{
                             expect(@(CGSizeEqualToSize(streamingLifecycleManager.screenSize, CGSizeZero))).to(beTrue());
                             expect(streamingLifecycleManager.dataSource).to(beNil());
                         });
                     });
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m]:
```suggestion

```

File: SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m [hunk 4]
```diff
@@ -583,37 +595,39 @@ __block void (^sendNotificationForHMILevel)(SDLHMILevel hmiLevel, SDLVideoStream
...
                             preferredResolutionHigh = [[SDLImageResolution alloc] initWithWidth:100 height:100];
-                            streamingLifecycleManager.preferredResolutions = @[preferredResolutionLow, preferredResolutionHigh];
-
                             streamingLifecycleManager.dataSource = testDataSource;
+                            streamingLifecycleManager.preferredResolutions = @[preferredResolutionLow, preferredResolutionHigh];
+                            
                             [streamingLifecycleManager handleProtocolStartServiceACKMessage:testVideoMessage];
                         });
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/DevAPISpecs/SDLStreamingVideoLifecycleManagerSpec.m]:
```suggestion

```

File: SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m [hunk 5]
```diff
@@ -272,41 +288,109 @@ __block void (^performTouchEvent)(SDLTouchManager* touchManager, SDLOnTouchEvent
...
-
+                    
                     performTouchEvent(touchManager, firstOnTouchEventStart);
                     performTouchEvent(touchManager, firstOnTouchEventEnd);
-
+                    
                     expectedDidCallSingleTap = YES;
                     expectedNumTimesHandlerCalled = 2;
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m]:
```suggestion

```

File: SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m [hunk 6]
```diff
@@ -272,41 +288,109 @@ __block void (^performTouchEvent)(SDLTouchManager* touchManager, SDLOnTouchEvent
...
+                        
                         expect(touchManagerCallback).to(equal(touchManager));
                         expect(@(CGPointEqualToPoint(point, controlPoint))).to(beTruthy());
                     };
-
+                    
                     performTouchEvent(touchManager, firstOnTouchEventStart);
                     performTouchEvent(touchManager, firstOnTouchEventEnd);
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m]:
```suggestion

```

File: SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m [hunk 7]
```diff
@@ -272,41 +288,109 @@ __block void (^performTouchEvent)(SDLTouchManager* touchManager, SDLOnTouchEvent
...
                         CGPoint point;
                         [invocation getArgument:&touchManagerCallback atIndex:2];
                         [invocation getArgument:&point atIndex:4];
-
+                        
                         expect(touchManagerCallback).to(equal(touchManager));
                         expect(@(CGPointEqualToPoint(point, controlPoint))).to(beTruthy());
                     };
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m]:
```suggestion

```

File: SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m [hunk 8]
```diff
@@ -272,41 +288,109 @@ __block void (^performTouchEvent)(SDLTouchManager* touchManager, SDLOnTouchEvent
...
+                __block CGPoint expectedScaledPoint = CGPointZero;
+                
                 it(@"should correctly handle a single tap", ^{
                     singleTapTests = ^(NSInvocation* invocation) {
                         __unsafe_unretained SDLTouchManager* touchManagerCallback;
                         CGPoint point;
                         [invocation getArgument:&touchManagerCallback atIndex:2];
                         [invocation getArgument:&point atIndex:4];
-
+
```
Comment (joeljfischer) [inline SmartDeviceLinkTests/UtilitiesSpecs/Touches/SDLTouchManagerSpec.m]:
```suggestion

```

File: SmartDeviceLink/SDLFocusableItemLocator.m [hunk 9]
```diff
@@ -73,6 +81,9 @@ - (void)sdl_parseViewHierarchy:(UIView *)currentView {
         return;
     }
 
+    // Force the view to update autolayout constraints. Otherwise the view's frame will not be correct if the root view controller's frame was resized.
+    [currentView layoutSubviews];
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Renamed properties in the `SDLStreamingVideoScaleManager`. `screenSize` is now `displayViewportResolution`.

File: SmartDeviceLink/SDLVideoStreamingCapability.h [hunk 10]
```diff
@@ -48,6 +71,30 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nullable, strong, nonatomic) NSNumber<SDLBool> *hapticSpatialDataSupported;
 
+/**
+ The diagonal screen size in inches.
+ 
+ Float, Optional, minvalue="0"
+ @since SDL 6.0
+ */
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *diagonalScreenSize;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLVideoStreamingCapability.h]: Renamed properties in the SDLStreamingVideoScaleManager. screenSize is now displayViewportResolution.

File: SmartDeviceLink/SDLStreamingMediaManagerConstants.m [hunk 11]
```diff
@@ -9,8 +9,6 @@
 #import <UIKit/UIKit.h>
 #import "SDLStreamingMediaManagerConstants.h"
 
-CGSize const SDLDefaultScreenSize = {0, 0};
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingMediaManagerConstants.m]: Fixed. Removed from the `.h`. The `SDLStreamingVideoScaleManager` now handles setting the the default screen size.

File: SmartDeviceLink/SDLFocusableItemLocator.m [hunk 12]
```diff
@@ -25,29 +27,35 @@ @interface SDLFocusableItemLocator()
 @property (nonatomic, strong) NSMutableArray<UIView *> *focusableViews;
 
 /**
- reference to SDLConnectionManager
+ Reference to SDLConnectionManager
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Fixed. Removed

File: SmartDeviceLink/SDLFocusableItemLocator.m [hunk 13]
```diff
@@ -25,29 +27,35 @@ @interface SDLFocusableItemLocator()
...
+    _videoScaleManager = videoScaleManager;
     _enableHapticDataRequests = NO;
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_projectionViewUpdated:) name:SDLDidUpdateProjectionView object:nil];
 
     return self;
 }
 
 - (void)updateInterfaceLayout {
+    // Adjust the root view controller frame
+    self.viewController.view.frame = self.videoScaleManager.screenFrame;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Fixed. I was trying to fix the fact that `rootViewController` is resized in the `SDLCarWindow` class after the `focusableViews` are parsed in the `SDLFocusableItemLocator` class. 

If the `viewController` is resized and it has auto layout constraints then the haptic rects sent by this class and the tapped views will be incorrect due to to the fact that resizing the `CarWindow``rootViewController` frame must done on the main thread. This means that the focusable views are calculated and sent **before** the `rootViewController` is resized in the `CarWindow` class. 

A workaround is to only send the `SDLDidUpdateProjectionView` notifications in the override `viewDidLayoutSubviews` of the set `rootViewController` (https://github.com/smartdevicelink/sdl_app_library_guides/issues/155).

File: SmartDeviceLink/SDLFocusableItemLocator.m (hunk 9 above)
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Fixed. Removed.

File: SmartDeviceLink/SDLFocusableItemLocator.m [hunk 14]
```diff
@@ -148,6 +160,10 @@ - (nullable UIView *)viewForPoint:(CGPoint)point {
  @param notification object with notification data
  */
 - (void)sdl_projectionViewUpdated:(NSNotification *)notification {
+    [self sdl_updateInterfaceLayout];
+}
+
+- (void)sdl_updateInterfaceLayout {
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Fixed. Removed.

File: SmartDeviceLink/SDLFocusableItemLocator.m [hunk 15]
```diff
@@ -157,6 +173,19 @@ - (void)sdl_projectionViewUpdated:(NSNotification *)notification {
...
+#pragma mark - Setters
+
+/**
+ Updates the interface layout when the scale or screen size changes.
+*/
+- (void)setVideoScaleManager:(SDLStreamingVideoScaleManager *)videoScaleManager {
+    SDLStreamingVideoScaleManager *oldScaleManager = self.videoScaleManager;
+    _videoScaleManager = videoScaleManager;
+
+    if (oldScaleManager.scale == videoScaleManager.scale && CGSizeEqualToSize(oldScaleManager.screenSize, videoScaleManager.screenSize)) { return; }
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Fixed. Removed

File: SmartDeviceLink/SDLFocusableItemLocatorType.h [hunk 16]
```diff
@@ -32,13 +34,15 @@ NS_ASSUME_NONNULL_BEGIN
...
  */
-- (instancetype)initWithViewController:(UIViewController *)viewController connectionManager:(id<SDLConnectionManagerType>)connectionManager;
+- (instancetype)initWithViewController:(UIViewController *)viewController connectionManager:(id<SDLConnectionManagerType>)connectionManager videoScaleManager:(SDLStreamingVideoScaleManager *)videoScaleManager;
 
 /**
  updateInterfaceLayout crawls through the view hierarchy, updates and keep tracks of views to be reported through Haptic RPC. This function is automatically called when SDLDidUpdateProjectionView notification is sent by the application.
  */
 - (void)updateInterfaceLayout;
 
+@property (nonatomic, strong) SDLStreamingVideoScaleManager *videoScaleManager;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocatorType.h]: Fixed. Removed. Added the `SDLStreamingVideoLifecycleManager` to the `init` for this class.

File: SmartDeviceLink/SDLFocusableItemLocator.h [hunk 17]
```diff
@@ -21,6 +23,8 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, strong) UIViewController *viewController;
 
+@property (nonatomic, strong) SDLStreamingVideoScaleManager *videoScaleManager;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.h]: Fixed. Removed

File: SmartDeviceLink/SDLFocusableItemLocator.m [hunk 18]
```diff
@@ -105,14 +116,15 @@ - (void)sdl_sendHapticRPC {
...
-
     for (UIView *view in self.focusableViews) {
         CGPoint originOnScreen = [self.viewController.view convertPoint:view.frame.origin toView:nil];
         CGRect convertedRect = {originOnScreen, view.bounds.size};
-        SDLRectangle* rect = [[SDLRectangle alloc] initWithCGRect:(convertedRect)];
+        SDLRectangle *rect = [[SDLRectangle alloc] initWithCGRect:convertedRect];
         // using the view index as the id field in SendHapticData request (should be guaranteed unique)
         NSUInteger rectId = [self.focusableViews indexOfObject:view];
         SDLHapticRect *hapticRect = [[SDLHapticRect alloc] initWithId:(UInt32)rectId rect:rect];
+        hapticRect = [self.videoScaleManager scaleHapticRect:hapticRect.copy];
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Fixed

File: SmartDeviceLink/SDLStreamingVideoLifecycleManager.m [hunk 19]
```diff
@@ -733,6 +736,12 @@ - (void)sdl_requestVideoCapabilities:(SDLVideoCapabilityResponseHandler)response
 
         SDLVideoStreamingCapability *videoCapability = ((SDLGetSystemCapabilityResponse *)response).systemCapability.videoStreamingCapability;
         SDLLogD(@"Video capabilities response received: %@", videoCapability);
+
+        float newScale = (videoCapability != nil && videoCapability.scale != nil) ? videoCapability.scale.floatValue : (float)0.0;
+        self.videoScaleManager.scale = newScale;
+        self.touchManager.scale = newScale;
+        self.focusableItemManager.videoScaleManager = [[SDLStreamingVideoScaleManager alloc] initWithScale:@(newScale) screenSize:self.screenSize];
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoLifecycleManager.m]: Fixed. The `SDLStreamingVideoLifecycleManager` now owns `videoScaleManager`.

File: SmartDeviceLink/SDLStreamingVideoLifecycleManager.m [hunk 20]
```diff
@@ -819,6 +828,12 @@ - (NSString *)videoStreamBackgroundString {
     return [NSString stringWithFormat:@"When it is safe to do so, open %@ on your phone", self.appName];
 }
 
+- (void)setScreenSize:(CGSize)screenSize {
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoLifecycleManager.m]: The `SDLStreamingVideoScaleManager` now owns the `screenSize` (renamed to `displayViewportResolution`)

File: SmartDeviceLink/SDLStreamingVideoScaleManager.h [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+ */
+@interface SDLStreamingVideoScaleManager : NSObject
+
+/**
+ The scale factor value to scale coordinates from one coordinate space to another.
+ */
+@property (assign, nonatomic) float scale;
+
+/**
+ The current screen size of the connected display
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.h]: Fixed. Added.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.h [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+ */
+@property (assign, nonatomic) CGSize screenSize;
+
+/**
+ The scaled frame for the view being streamed to the connected display.
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.h]: Fixed. Renamed to `appViewportFrame` and added better documentation.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.h [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+*/
+@property (assign, nonatomic, readonly) CGRect screenFrame;
+
+- (instancetype)init NS_UNAVAILABLE;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.h]: Fixed. Removed `defaultConfiguration`

File: SmartDeviceLink/SDLStreamingVideoScaleManager.h [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+
+/**
+ Creates a default streaming video scale manager.
+
+ @return A default configuration that may be customized.
+ */
++ (instancetype)defaultConfiguration;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.h]: Fixed. Removed.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.h [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+
+/**
+ Convenience init for creating a scale manager with the scale and connected display screen size.
+
+ @param scale The scale factor value to scale coordinates from one coordinate space to another
+ @param screenSize The current screen size of the connected display
+ @return A SDLStreamingVideoScaleManager object
+*/
+- (instancetype)initWithScale:(nullable NSNumber *)scale screenSize:(CGSize)screenSize;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.h]: Fixed. Changed data type to `float`

File: SmartDeviceLink/SDLStreamingVideoScaleManager.m [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+    return [[self.class alloc] initWithScale:@(DefaultScaleValue) screenSize:SDLDefaultScreenSize];
+}
+
+- (instancetype)initWithScale:(nullable NSNumber *)scale screenSize:(CGSize)screenSize {
+    self = [super init];
+    if (!self) {
+        return nil;
+    }
+
+    _scale = [self.class validateScale:scale.floatValue];
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.m]: Fixed.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.m [hunk 27]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+#pragma mark - Getters and Setters
+
+- (CGRect)screenFrame {
+    // Screen capture in the CarWindow API only works if the width and height are integer values
+    return CGRectMake(0,
+                      0,
+                      roundf((float)self.screenSize.width / self.scale),
+                      roundf((float)self.screenSize.height / self.scale));
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.m]: Fixed

File: SmartDeviceLink/SDLTouchManager.h [hunk 28]
```diff
@@ -78,6 +78,11 @@ typedef void(^SDLTouchEventHandler)(SDLTouch *touch, SDLTouchType type);
  */
 @property (nonatomic, assign, getter=isTouchEnabled) BOOL touchEnabled;
 
+/**
+ The scale factor value to scale coordinates from one coordinate space to another.
+ */
+@property (nonatomic, assign) float scale;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLTouchManager.h]: Fixed.

File: SmartDeviceLink/SDLTouchManager.m [hunk 29]
```diff
@@ -98,17 +97,21 @@ @interface SDLTouchManager ()
  */
 @property (nonatomic) CGPoint lastNotifiedTouchLocation;
 
+
+@property (strong, nonatomic) SDLStreamingVideoScaleManager *videoScaleManager;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLTouchManager.m]: Fixed.

File: SmartDeviceLink/SDLTouchManager.m [hunk 30]
```diff
@@ -184,7 +187,8 @@ - (void)sdl_onTouchEvent:(SDLRPCNotificationNotification *)notification {
         return;
     }
 
-    SDLOnTouchEvent* onTouchEvent = (SDLOnTouchEvent*)notification.notification;
+    SDLOnTouchEvent *onTouchEvent = (SDLOnTouchEvent *)notification.notification;
+    onTouchEvent = [self.videoScaleManager scaleTouchEventCoordinates:onTouchEvent.copy];
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLTouchManager.m]: Fixed.

File: SmartDeviceLink/SDLTouchManager.m [hunk 31]
```diff
@@ -493,6 +497,11 @@ - (void)sdl_cancelSingleTapTimer {
     self.singleTapTimer = nil;
 }
 
+- (void)setScale:(float)scale {
+    _scale = scale;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLTouchManager.m]: Fixed. Removed.

File: SmartDeviceLink/SDLFocusableItemLocator.m (hunk 13 above)
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLFocusableItemLocator.m]: Although it looks like if the `carWindowRenderingType` is set to `SDLCarWindowRenderingTypeViewAfterScreenUpdates`, the `viewDidLayoutSubviews` is called every time `syncFrame` is called. Works fine if set to `SDLCarWindowRenderingTypeLayer`

File: SmartDeviceLinkTests/SDLStreamingVideoScaleManagerSpec.m [hunk 32]
```diff
@@ -0,0 +1,135 @@
...
+    __block CGSize testScreenSize = CGSizeMake(200, 400);
+
+    it(@"should properly init a default configuration", ^{
+        videoScaleManager = [[SDLStreamingVideoScaleManager alloc] init];
+
+        expect(@(videoScaleManager.scale)).to(equal(1.0));
+        expect(CGSizeEqualToSize(videoScaleManager.displayViewportResolution, CGSizeZero)).to(beTrue());
+    });
+
+    it(@"should properly init a default configuration", ^{
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLinkTests/SDLStreamingVideoScaleManagerSpec.m]: Fixed.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.m [hunk 33]
```diff
@@ -0,0 +1,97 @@
...
+
+const float DefaultScaleValue = 1.0;
+CGSize const SDLDefaultDisplayViewportResolution = {0, 0};
+
+- (instancetype)init {
+    return [[self.class alloc] initWithScale:DefaultScaleValue displayViewportResolution:SDLDefaultDisplayViewportResolution];
+}
+
+- (void)stop {
+    self.displayViewportResolution = SDLDefaultDisplayViewportResolution;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.m]: Yep. Fixed.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.m [hunk 34]
```diff
@@ -0,0 +1,97 @@
...
+@interface SDLStreamingVideoScaleManager ()
+
+@property (assign, nonatomic, readwrite) CGRect appViewportFrame;
+
+@end
+
+@implementation SDLStreamingVideoScaleManager
+
+const float DefaultScaleValue = 1.0;
+CGSize const SDLDefaultDisplayViewportResolution = {0, 0};
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.m]: Fixed.

File: SmartDeviceLink/SDLStreamingVideoScaleManager.h [hunk 35]
```diff
@@ -0,0 +1,75 @@
...
+ */
+@property (assign, nonatomic) CGSize displayViewportResolution;
+
+/**
+ The frame of the app's projecting view. This is calculated by dividing the display's viewport resolution by the scale. The units are points.
+ */
+@property (assign, nonatomic, readonly) CGRect appViewportFrame;
+
+/**
+ Creates a default streaming video scale manager.
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLStreamingVideoScaleManager.h]: Fixed. Added a description fo the default parameters.

File: SmartDeviceLink/SDLTouchManager.h [hunk 36]
```diff
@@ -96,6 +97,15 @@ typedef void(^SDLTouchEventHandler)(SDLTouch *touch, SDLTouchType type);
  */
 - (instancetype)initWithHitTester:(nullable id<SDLFocusableItemHitTester>)hitTester;
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLTouchManager.h]: Fixed.

File: SmartDeviceLink-iOS.podspec [hunk 37]
```diff
@@ -365,6 +365,7 @@ ss.public_header_files = [
 'SmartDeviceLink/SDLStreamingMediaManager.h',
 'SmartDeviceLink/SDLStreamingMediaManagerConstants.h',
 'SmartDeviceLink/SDLStreamingMediaManagerDataSource.h',
+'SmartDeviceLink/SDLStreamingVideoScaleManager.h',
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink-iOS.podspec]: Created issue #1432

**Pull request merged**